From 682b67de8988d8b1ca24af64c960798fd5acb6b4 Mon Sep 17 00:00:00 2001
From: dtodoroff <d.todorov@vekatech.com>
Date: Tue, 15 Jul 2025 10:06:02 +0300
Subject: [PATCH] add vkboards support


diff --git a/dgmodul4.c b/dgmodul4.c
index ab37074..cb06274 100644
--- a/dgmodul4.c
+++ b/dgmodul4.c
@@ -53,8 +53,8 @@ static uint32_t CheckQspiFlashId(void)
 	gDeviceId		= deviceId;
 	switch(manuId)
 	{
-		case CYPRESS_MANUFACTURER_ID:
-			PutStr(" Cypress : ", 0);
+		case INFINEON_MANUFACTURER_ID:
+			PutStr(" Infineon : ", 0);
 			switch(deviceId)
 			{
 				case DEVICE_ID_S25FS512S:
@@ -67,6 +67,11 @@ static uint32_t CheckQspiFlashId(void)
 						gQspi_sa_size    = SA_256KB;
 						gQspi_end_addess = TOTAL_SIZE_16MB - 0x8000 - 1;
 				break;
+                case DEVICE_ID_S25FL128L:
+                        PutStr("S25FL128L", 1);
+                        gQspi_sa_size    = SA_64KB;
+                        gQspi_end_addess = TOTAL_SIZE_16MB - 0x8000 - 1;
+                break;
 				default:
 					ret = 1;
 				break;
@@ -132,8 +137,8 @@ static uint32_t CheckQspiFlashId(void)
 			PutStr(" Macronix : ", 0);
 			switch(deviceId)
 			{
-				case DEVICE_ID_MX25L12805:
-						PutStr("MX25L12805", 1);
+				case DEVICE_ID_MX25L12833F:
+						PutStr("MX25L12833F", 1);
 						gQspi_sa_size    = SA_64KB;
 						gQspi_end_addess = TOTAL_SIZE_16MB - 0x8000 - 1;
 				break;
@@ -147,8 +152,8 @@ static uint32_t CheckQspiFlashId(void)
 						gQspi_sa_size    = SA_64KB;
 						gQspi_end_addess = TOTAL_SIZE_64MB - 0x8000 - 1;
 				break;
-				case DEVICE_ID_MX66U25635F:
-						PutStr("MX66U25635F", 1);
+				case DEVICE_ID_MX25U25645G:
+						PutStr("MX25U25645G", 1);
 						gQspi_sa_size    = SA_64KB;
 						gQspi_end_addess = TOTAL_SIZE_32MB - 0x8000 - 1;
 				break;
diff --git a/include/dgmodul4.h b/include/dgmodul4.h
index 2e20529..21b10a8 100644
--- a/include/dgmodul4.h
+++ b/include/dgmodul4.h
@@ -34,7 +34,7 @@
 #define	TOTAL_SIZE_16MB			0x01000000
 #define	TOTAL_SIZE_8MB			0x00800000
 
-#define	CYPRESS_MANUFACTURER_ID		0x01	/* Cypress	*/
+#define	INFINEON_MANUFACTURER_ID	0x01	/* Infineon	*/
 #define	WINBOND_MANUFACTURER_ID		0xEF	/* Winbond	*/
 #define	MACRONIX_MANUFACTURER_ID	0xC2	/* Macronix	*/
 #define	MICRON_MANUFACTURER_ID		0x20	/* Micron	*/
@@ -42,6 +42,7 @@
 
 #define	DEVICE_ID_S25FS128S		0x2018
 #define	DEVICE_ID_S25FS512S		0x0220
+#define	DEVICE_ID_S25FL128L  	0x6018
 
 #define	DEVICE_ID_W25Q64JV		0x4017
 #define	DEVICE_ID_W25Q64JW		0x6017
@@ -53,10 +54,10 @@
 #define	DEVICE_ID_W25Q512JV		0x4020
 #define	DEVICE_ID_W25Q512JV_DTR		0x7020
 
-#define	DEVICE_ID_MX25L12805		0x2018
+#define	DEVICE_ID_MX25L12833F		0x2018
 #define	DEVICE_ID_MX25L25645G		0x2019
 #define	DEVICE_ID_MX25L51245G		0x201A
-#define	DEVICE_ID_MX66U25635F		0x2539
+#define	DEVICE_ID_MX25U25645G		0x2539
 #define	DEVICE_ID_MX66U51235F		0x253A
 #define	DEVICE_ID_MX66UM1G45G		0x803B
 #define	DEVICE_ID_MX66UW1G45G		0x813B
diff --git a/makefile b/makefile
index 0d4c834..0a56289 100644
--- a/makefile
+++ b/makefile
@@ -44,6 +44,15 @@ DEVICE   = RZG2L
 DDR_TYPE = DDR4
 DDR_SIZE = 4GB
 SWIZZLE  = T2C
+else ifeq ("$(BOARD)", "VKRZG2L")
+#--------------------------------------
+# RZ/G2L VK-RZ/G2L board
+#--------------------------------------
+FILENAME_ADD = _VKRZG2L
+DEVICE   = RZG2L
+DDR_TYPE = DDR4
+DDR_SIZE = 2GB_1PCS
+SWIZZLE  = T1BC
 else ifeq ("$(BOARD)", "RZG2LC_SMARC")
 #--------------------------------------
 # RZ/G2LC Smarc board
@@ -62,6 +71,33 @@ DEVICE   = RZG2LC
 DDR_TYPE = DDR3L
 DDR_SIZE = 1GB
 SWIZZLE  = T3CL
+else ifeq ("$(BOARD)", "VKRZG2LC")
+#--------------------------------------
+# VK-RZ/G2LC board
+#--------------------------------------
+FILENAME_ADD = _VKRZG2LC
+DEVICE   = RZG2LC
+DDR_TYPE = DDR4
+DDR_SIZE = 2GB_1PCS
+SWIZZLE  = T3BC
+else ifeq ("$(BOARD)", "VKCMG2LC")
+#--------------------------------------
+# VK-CMRZ/G2LC board
+#--------------------------------------
+FILENAME_ADD = _VKCMG2LC
+DEVICE   = RZG2LC
+DDR_TYPE = DDR4
+DDR_SIZE = 2GB_1PCS
+SWIZZLE  = T3BC
+else ifeq ("$(BOARD)", "VKCMG2LC_1")
+#--------------------------------------
+# VK-CMRZ/G2LC board (1GB DDR4)
+#--------------------------------------
+FILENAME_ADD = _VKCMG2LC
+DEVICE   = RZG2LC
+DDR_TYPE = DDR4
+DDR_SIZE = 1GB_1PCS
+SWIZZLE  = T3BC
 else ifeq ("$(BOARD)", "RZG2UL_TYPE1_DEV")
 #--------------------------------------
 # RZ/G2UL Type1 DDR4 Dev board
@@ -92,6 +128,16 @@ DDR_TYPE = DDR4
 DDR_SIZE = 1GB_1PCS
 SWIZZLE  = T3BCUD2
 DEVICE_TYPE = 1
+else ifeq ("$(BOARD)", "VKRZG2UL")
+#--------------------------------------
+# VK-RZ/G2UL board
+#--------------------------------------
+FILENAME_ADD = _VKRZG2UL
+DEVICE   = RZG2UL
+DDR_TYPE = DDR4
+DDR_SIZE = 1GB_1PCS
+SWIZZLE  = T3BCUD2
+DEVICE_TYPE = 1
 else ifeq ("$(BOARD)", "RZG2UL_TYPE2_SMARC")
 #--------------------------------------
 # RZ/G2UL Type2 Smarc board
@@ -148,6 +194,24 @@ DEVICE   = RZV2L
 DDR_TYPE = DDR4
 DDR_SIZE = 4GB
 SWIZZLE  = T2C
+else ifeq ("$(BOARD)", "VKRZV2L")
+#--------------------------------------
+# VK-RZV2L board
+#--------------------------------------
+FILENAME_ADD = _VKRZV2L
+DEVICE   = RZV2L
+DDR_TYPE = DDR4
+DDR_SIZE = 2GB_1PCS
+SWIZZLE  = T1BC
+else ifeq ("$(BOARD)", "VKCMV2L")
+#--------------------------------------
+# VK-CMV2L board
+#--------------------------------------
+FILENAME_ADD = _VKCMV2L
+DEVICE   = RZV2L
+DDR_TYPE = DDR4
+DDR_SIZE = 2GB_1PCS
+SWIZZLE  = T1BC
 else ifeq ("$(BOARD)", "RZA3UL_DEV")
 #--------------------------------------
 # RZ/A3UL DDR4 Dev board
diff --git a/makefile.linaro b/makefile.linaro
index 453572f..0a759e1 100644
--- a/makefile.linaro
+++ b/makefile.linaro
@@ -44,6 +44,15 @@ DEVICE   = RZG2L
 DDR_TYPE = DDR4
 DDR_SIZE = 4GB
 SWIZZLE  = T2C
+else ifeq ("$(BOARD)", "VKRZG2L")
+#--------------------------------------
+# RZ/G2L VK-RZ/G2L board
+#--------------------------------------
+FILENAME_ADD = _VKRZG2L
+DEVICE   = RZG2L
+DDR_TYPE = DDR4
+DDR_SIZE = 2GB_1PCS
+SWIZZLE  = T1BC
 else ifeq ("$(BOARD)", "RZG2LC_SMARC")
 #--------------------------------------
 # RZ/G2LC Smarc board
@@ -62,6 +71,33 @@ DEVICE   = RZG2LC
 DDR_TYPE = DDR3L
 DDR_SIZE = 1GB
 SWIZZLE  = T3CL
+else ifeq ("$(BOARD)", "VKRZG2LC")
+#--------------------------------------
+# VK-RZ/G2LC board
+#--------------------------------------
+FILENAME_ADD = _VKRZG2LC
+DEVICE   = RZG2LC
+DDR_TYPE = DDR4
+DDR_SIZE = 2GB_1PCS
+SWIZZLE  = T3BC
+else ifeq ("$(BOARD)", "VKCMG2LC")
+#--------------------------------------
+# VK-CMRZ/G2LC board
+#--------------------------------------
+FILENAME_ADD = _VKCMG2LC
+DEVICE   = RZG2LC
+DDR_TYPE = DDR4
+DDR_SIZE = 2GB_1PCS
+SWIZZLE  = T3BC
+else ifeq ("$(BOARD)", "VKCMG2LC_1")
+#--------------------------------------
+# VK-CMRZ/G2LC board (1GB DDR4)
+#--------------------------------------
+FILENAME_ADD = _VKCMG2LC
+DEVICE   = RZG2LC
+DDR_TYPE = DDR4
+DDR_SIZE = 1GB_1PCS
+SWIZZLE  = T3BC
 else ifeq ("$(BOARD)", "RZG2UL_TYPE1_DEV")
 #--------------------------------------
 # RZ/G2UL Type1 DDR4 Dev board
@@ -92,6 +128,16 @@ DDR_TYPE = DDR4
 DDR_SIZE = 1GB_1PCS
 SWIZZLE  = T3BCUD2
 DEVICE_TYPE = 1
+else ifeq ("$(BOARD)", "VKRZG2UL")
+#--------------------------------------
+# VK-RZ/G2UL board
+#--------------------------------------
+FILENAME_ADD = _VKRZG2UL
+DEVICE   = RZG2UL
+DDR_TYPE = DDR4
+DDR_SIZE = 1GB_1PCS
+SWIZZLE  = T3BCUD2
+DEVICE_TYPE = 1
 else ifeq ("$(BOARD)", "RZG2UL_TYPE2_SMARC")
 #--------------------------------------
 # RZ/G2UL Type2 Smarc board
@@ -148,6 +194,24 @@ DEVICE   = RZV2L
 DDR_TYPE = DDR4
 DDR_SIZE = 4GB
 SWIZZLE  = T2C
+else ifeq ("$(BOARD)", "VKRZV2L")
+#--------------------------------------
+# VK-RZV2L board
+#--------------------------------------
+FILENAME_ADD = _VKRZV2L
+DEVICE   = RZV2L
+DDR_TYPE = DDR4
+DDR_SIZE = 2GB_1PCS
+SWIZZLE  = T1BC
+else ifeq ("$(BOARD)", "VKCMV2L")
+#--------------------------------------
+# VK-CMV2L board
+#--------------------------------------
+FILENAME_ADD = _VKCMV2L
+DEVICE   = RZV2L
+DDR_TYPE = DDR4
+DDR_SIZE = 2GB_1PCS
+SWIZZLE  = T1BC
 else ifeq ("$(BOARD)", "RZA3UL_DEV")
 #--------------------------------------
 # RZ/A3UL DDR4 Dev board
diff --git a/spiflash1drv.c b/spiflash1drv.c
index ca6b172..04d7286 100644
--- a/spiflash1drv.c
+++ b/spiflash1drv.c
@@ -366,7 +366,7 @@ void SectorEraseQspi_Flash(uint32_t EraseStatAdd, uint32_t EraseEndAdd)
 	PutStr(str, 1);
 #endif
 
-	if ((gManufacturerId == CYPRESS_MANUFACTURER_ID ) && (gDeviceId == DEVICE_ID_S25FS128S))
+	if ((gManufacturerId == INFINEON_MANUFACTURER_ID ) && (gDeviceId == DEVICE_ID_S25FS128S))
 	{
 		if (gQspi_sa_size == SA_256KB)
 		{
-- 
2.43.0

