From f5b6afebb7823a83d01643ce920be0c418eedecc Mon Sep 17 00:00:00 2001
From: dtodoroff <d.todorov@vekatech.com>
Date: Thu, 17 Jul 2025 10:12:56 +0300
Subject: [PATCH] modify bptool mode names


diff --git a/tools/renesas/rz_boot_param/bptool.c b/tools/renesas/rz_boot_param/bptool.c
index 675c8d4bc..7bd7968c7 100644
--- a/tools/renesas/rz_boot_param/bptool.c
+++ b/tools/renesas/rz_boot_param/bptool.c
@@ -43,10 +43,10 @@ static int get_loadinfo(char *mode, uint32_t *ldr, uint32_t *bpn)
 {
 	int err = 0;
 
-	if (strcmp(mode, "spi") == 0) {
+	if (strcmp(mode, "qspi") == 0) {
 		*ldr = SPI_LOAD_OFFSET;
 		*bpn = 1;
-	} else if (strcmp(mode, "mmc") == 0) {
+	} else if (strcmp(mode, "emmc") == 0) {
 		*ldr = MMC_LOAD_OFFSET;
 		*bpn = 1;
 	} else if (strcmp(mode, "scif") == 0) {
@@ -67,7 +67,7 @@ int main(int argc, char *argv[])
 	int err = -1;
 	FILE *fp = NULL;
 
-	char *mode = "spi";
+	char *mode = "qspi";
 	uint32_t i;
 	uint32_t load_offset_adr = 0;
 	uint32_t dest_offset_adr = 0;
@@ -83,7 +83,7 @@ int main(int argc, char *argv[])
 		printf("\t<BOOT_PARAM>     Output Boot parameter file path.\n");
 		printf("\t<DEST_ADDR>      Destination address in hexadecimal.\n");
 		printf("\t[BOOT_MODE]      Boot mode selection (default: spi).\n");
-		printf("\t                 - \"spi\", \"mmc\", \"scif\", \"esd\"\n");
+		printf("\t                 - \"qspi\", \"emmc\", \"scif\", \"esd\"\n");
 		goto exit;
 	}
 
-- 
2.43.0

