From 67d432afa3b19df330632df4336b5dbf7baa4ad0 Mon Sep 17 00:00:00 2001
From: dtodoroff <d.todorov@vekatech.com>
Date: Tue, 15 Jul 2025 15:56:15 +0300
Subject: [PATCH 4/8] add option rawimg for fastboot flash write command


diff --git a/drivers/fastboot/fb_mmc.c b/drivers/fastboot/fb_mmc.c
index cbb3f7b1de..6f82eb00ee 100644
--- a/drivers/fastboot/fb_mmc.c
+++ b/drivers/fastboot/fb_mmc.c
@@ -514,6 +514,60 @@ void fastboot_mmc_flash_write(const char *cmd, void *download_buffer,
 	struct blk_desc *dev_desc;
 	struct disk_partition info = {0};
 
+	if (strncasecmp(cmd, "rawimg", 6) == 0) {
+		dev_desc = fastboot_mmc_get_dev(response);
+		if (!dev_desc)
+			return;
+
+		if (is_sparse_image(download_buffer)) {
+			struct fb_mmc_sparse  sparse_priv;
+			static struct sparse_storage sparse;
+			int err;
+			int header_size=sizeof(sparse_header_t)+sizeof(chunk_header_t);
+
+			/* Only 1st fragment packet got MBR information */
+			if ( 0 == is_valid_dos_buf(download_buffer+header_size)) {
+				printf("Found valid MBR Information in image.\n");
+				sparse.start = 0;
+			}
+
+			sparse_priv.dev_desc = dev_desc;
+
+			sparse.blksz = dev_desc->blksz; /* eMMC block size: 512 */
+			sparse.size = dev_desc->lba;    /* eMMC total blocks: 30621696 * 512/1024/1024/1024= 14.6GB */
+			sparse.write = fb_mmc_sparse_write;
+			sparse.reserve = fb_mmc_sparse_reserve;
+			sparse.mssg = fastboot_fail;
+
+			/* write_sparse_image() will update sparse.start */
+			sparse.priv = &sparse_priv;
+			err = write_sparse_image(&sparse, cmd, download_buffer, response);
+			if (!err)
+				fastboot_okay("Flashing raw image Okay", response);
+		} else {
+			long start = 0;
+			int blksz = dev_desc->blksz;
+			lbaint_t blkcnt;
+			lbaint_t blks;
+
+			/* determine number of blocks to write */
+			blkcnt = ((download_bytes + (blksz - 1)) & ~(blksz - 1));
+			blkcnt = blkcnt / blksz;
+
+			blks = fb_mmc_blk_write(dev_desc, start / blksz, blkcnt, download_buffer);
+			if (blks != blkcnt) {
+				printf("........ wrote " LBAFU " bytes to 0x%lx failed.\n", blkcnt * blksz, start);
+				fastboot_fail("Flashing raw image failed", response);
+				return;
+			}
+
+			printf("........ wrote " LBAFU " bytes to 0x%lx success.\n", blkcnt * blksz, start);
+			fastboot_okay("Flashing raw image Okay", response);
+		}
+
+		return ;
+	}
+
 #ifdef CONFIG_FASTBOOT_MMC_BOOT_SUPPORT
 	if (strcmp(cmd, CONFIG_FASTBOOT_MMC_BOOT1_NAME) == 0) {
 		dev_desc = fastboot_mmc_get_dev(response);
-- 
2.43.0

